internal/measurement/measurement.go

```diff
@@ -172,69 +176,55 @@ func Percentage(value, total int64) string {
 	}
 }
 
+// unit includes a list of names representing a specific unit and a factor
+// which one can multiple a value in the specified unit by to get the value
+// in terms of the base unit.
+type unit struct {
+	preferredName string
+	names         []string
+	factor        float64
+}
+
+var memoryUnits = []unit{
+	{"B", []string{"b", "byte"}, 1},
+	{"kB", []string{"kb", "kbyte", "kilobyte"}, 1024},
+	{"MB", []string{"mb", "mbyte", "megabyte"}, 1024 * 1024},
+	{"GB", []string{"gb", "gbyte", "gigabyte"}, 1024 * 1024 * 1024},
+	{"TB", []string{"tb", "tbyte", "terabyte"}, 1024 * 1024 * 1024 * 1024},
+	{"PB", []string{"pb", "pbyte", "petabyte"}, 1024 * 1024 * 1024 * 1024 * 1024},
+}
+
 // isMemoryUnit returns whether a name is recognized as a memory size
 // unit.
 func isMemoryUnit(unit string) bool {
-	switch strings.TrimSuffix(strings.ToLower(unit), "s") {
-	case "byte", "b", "kilobyte", "kb", "megabyte", "mb", "gigabyte", "gb":
```

Comment:
Interesting that terabyte wasn't recognized as a memory unit?

Reply:
I think terabyte was a valid unit for the purposes of display, but a unit that probably wouldn't have been used in the profile.proto (since that represents data in integers, and 1TiB as the smallest possible value is pretty coarse). 

But this also did surprise me.

Marking as resolved.